Update OpenStack client installation (#81)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_validating-clientside-installation.adoc

[id="validating-clientside-installation_{context}"]
= Validating client-side installation

Data collection can be validated from the {ProjectShort} storage domain by querying the data sources for delivered data. To validate individual nodes in the {OpenStack} deployment first connect to the console via SSH.

> . Login to an overcloud node, for example controller-0

Reviewer: ```suggestion
. Log in to an overcloud node, for example, controller-0:
```

> . Check that `metrics_qdr` container is running on the node:

Reviewer: ```suggestion
. Ensure that `metrics_qdr` container is running on the node:
```

+
----
$ sudo podman container inspect --format '{{.State.Status}}' metrics_qdr

running
----

> . Return the internal network address that QDR is running on, for example `172.17.1.44` listening on port `5666`:

Reviewer: ```suggestion
. Return the internal network address on which {MessageBus} is running, for example, `172.17.1.44` listening on port `5666`:
```

+
----
$ sudo podman exec -it metrics_qdr cat /etc/qpid-dispatch/qdrouterd.conf

listener {
    host: 172.17.1.44
    port: 5666
    authenticatePeer: no
    saslMechanisms: ANONYMOUS
}
----

> . Return a list of connections to the local QDR:

Reviewer: ```suggestion
. Return a list of connections to the local {MessageBus}:
```

  1    stf-default-interconnect-5671-service-telemetry.apps.stfcloudops1.lab.upshift.rdu2.redhat.com:443  stf-default-interconnect-7458fd4d69-bgzfb                                                                  edge    out  TLSv1.2(DHE-RSA-AES256-GCM-SHA384)  anonymous-user
  12   172.17.1.44:60290                                                                                  openstack.org/om/container/controller-0/ceilometer-agent-notification/25/5c02cee550f143ec9ea030db5cccba14  normal  in   no-security                         no-auth
  16   172.17.1.44:36408                                                                                  metrics                                                                                                    normal  in   no-security                         anonymous-user
  899  172.17.1.44:39500                                                                                  10a2e99d-1b8a-4329-b48c-4335e5f75c84                                                                       normal  in   no-security                         no-auth
----

> We expect to see four connections:

Reviewer: ```suggestion
There are four connections:
```

. Outbound connection to STF
. Inbound connection from collectd
. Inbound connection from ceilometer
. Inbound connection from our `qdstat` client

> The outbound STF connection will be to the host provided to the `MetricsQdrConnectors` host parameter and will be the route for the STF storage domain. The other hosts will be internal network addresses of the client connections to this QDR.

Reviewer: ```suggestion
Provide the host value of the outbound {ProjectShort} connection to the `MetricsQdrConnectors` host parameter. This will be the route for the {ProjectShort} storage domain. The other hosts are internal network addresses of the client connections to this {MessageBus}.
```
Should this be a step like in my suggested edit? Or is this just information for the user to hold on to for 
```suggestion
The outbound {ProjectShort} connection is provided to the `MetricsQdrConnectors` host parameter and is the route for the  {ProjectShort} storage domain. The other hosts are internal network addresses of the client connections to this {MessageBus}.
```

> The outbound STF connection will be to the host provided to the `MetricsQdrConnectors` host parameter and will be the route for the STF storage domain. The other hosts will be internal network addresses of the client connections to this QDR.

Reviewer: ```suggestion
The outbound {ProjectShort} connection is provided to the `MetricsQdrConnectors` host parameter and is the route for the {ProjectShort} storage domain. The other hosts are internal network addresses of the client connections to this {MessageBus}.
```

> . Next we can check that messages are being delivered by listing the links and checking the `_edge` address for delivery of messages by looking at the `deliv` column:

Reviewer: ```suggestion
. To ensure that messages are being delivered, list the links and view the `_edge` address in the `deliv` column for delivery of messages:
```

  endpoint  in   1        9                                              250  0    0      0       0        0        0       0    0    0    0    0      0
  endpoint  out  1        10                                             250  0    0      0       911      911      0       0    0    0    0    911    0
  endpoint  in   1        11                                             250  0    0      0       0        911      0       0    0    0    0    0      0
  endpoint  out  12       32          local   temp.lSY6Mcicol4J2Kp       250  0    0      0       0        0        0       0    0    0    0    0      0
  endpoint  in   16       41                                             250  0    0      0       2979924  2979924  0       0    0    0    0    0      0
  endpoint  in   912      1834        mobile  $management           0    250  0    0      0       1        0        0       1    0    0    0    0      0
  endpoint  out  912      1835        local   temp.9Ok2resI9tmt+CT       250  0    0      0       0        0        0       0    0    0    0    0      0
----

> To list the addresses from {OpenStack} nodes to {ProjectShort}, connect with the `oc` client to {OpenShiftShort} to get the {MessageBus} pod name then list the connections.

Reviewer: ```suggestion
To list the addresses from {OpenStack} nodes to {ProjectShort}, use the `oc` client to  connect to {OpenShiftShort} to get the {MessageBus} pod name and list the connections:
```

> . List the available QDR pods:

Reviewer: ```suggestion
. List the available {MessageBus} pods:
```

  3   10.130.0.51:43110  rcv[stf-default-collectd-notification-smartgateway-698c87fbb7-f28v6]  normal  in   no-security                             anonymous-user          000:21:36:53  006:21:49:09
  22  10.128.0.1:51948   Router.ceph-0.redhat.local                                            edge    in   TLSv1/SSLv3(DHE-RSA-AES256-GCM-SHA384)  anonymous-user          000:00:00:03  000:22:08:43
  23  10.128.0.1:51950   Router.compute-0.redhat.local                                         edge    in   TLSv1/SSLv3(DHE-RSA-AES256-GCM-SHA384)  anonymous-user          000:00:00:03  000:22:08:43
  24  10.128.0.1:52082   Router.controller-0.redhat.local                                      edge    in   TLSv1/SSLv3(DHE-RSA-AES256-GCM-SHA384)  anonymous-user          000:00:00:00  000:22:08:34
  27  127.0.0.1:42202    c2f541c1-4c97-4b37-a189-a396c08fb079                                  normal  in   no-security                             no-auth                 000:00:00:00  000:00:00:00
----
+

> In our case we see 3 `edge` connections from our {OpenStack} nodes (connection `id` 22, 23, and 24).

Reviewer: ```suggestion
In this example, there are three `edge` connections from our {OpenStack} nodes with connection `id` 22, 23, and 24.
```

> . If we return the addresses, we can see the number of messages delivered by the network:

Reviewer: ```suggestion
. To view the number of messages delivered by the network, use each address in the `oc exec` command:
```
